lib/gis/env.c

```diff
@@ -115,8 +115,8 @@ void G__read_gisrc_env(void)
 }
 
 static void parse_env(FILE *fd, int loc)
-{    
-    char buf[200];
+{
+    char buf[GPATH_MAX + strlen("GISDBASE: \0\n")];
```

Comment:
Just hardcode the length and add a comment explaining it.

Reply:
You mean hardcode `strlen("GISDBASE: \0\n")`, or the whole thing, GPATH_MAX + strlen("GISDBASE: \0\n"`?

I'm assuming the former.

Reply:
Not sure if you meant a more detailed comment; how about this one?

Reply:
something like `GPATH_MAX + 10`

general note: strlen only counts up till null character, `\0\n` is superfluous

Reply:
Ah, right. I added `+ 12` to account for the `\n` and the `\0` which are I think also returned in the buffer, so `GPATH_MAX + 12`.

Reply:
(But maybe my comment is incorrect then…)

Reply:
Please put the comment on a separate line above declaration of `char buf[GPATH_MAX + 12]` (with C style comment), eg.:

```
/* account for long lines up to GPATH_MAX  */
```
and why not round up the 12 chars to 16.

Reply:
Done. Sorry I forgot about the C-style comments. Not that often I touch pure C files these days.